src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetObjectForNativeVariantTests.cs

```diff
@@ -0,0 +1,169 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class GetObjectForNativeVariantTests
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        public struct Record
+        {
+            private IntPtr _record;
+            private IntPtr _recordInfo;
+        }
+
+        [StructLayout(LayoutKind.Explicit)]
+        public struct UnionTypes
+        {
+            [FieldOffset(0)] internal SByte _i1;
+            [FieldOffset(0)] internal Int16 _i2;
+            [FieldOffset(0)] internal Int32 _i4;
+            [FieldOffset(0)] internal Int64 _i8;
+            [FieldOffset(0)] internal Byte _ui1;
+            [FieldOffset(0)] internal UInt16 _ui2;
+            [FieldOffset(0)] internal UInt32 _ui4;
+            [FieldOffset(0)] internal UInt64 _ui8;
+            [FieldOffset(0)] internal Int32 _int;
+            [FieldOffset(0)] internal UInt32 _uint;
+            [FieldOffset(0)] internal Single _r4;
+            [FieldOffset(0)] internal Double _r8;
+            [FieldOffset(0)] internal Int64 _cy;
+            [FieldOffset(0)] internal double _date;
+            [FieldOffset(0)] internal IntPtr _bstr;
+            [FieldOffset(0)] internal IntPtr _unknown;
+            [FieldOffset(0)] internal IntPtr _dispatch;
+            [FieldOffset(0)] internal IntPtr _pvarVal;
+            [FieldOffset(0)] internal IntPtr _byref;
+            [FieldOffset(0)] internal Record _record;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct TypeUnion
+        {
+            public ushort vt;
+            public ushort wReserved1;
+            public ushort wReserved2;
+            public ushort wReserved3;
+            public UnionTypes _unionTypes;
+        }
+
+        [StructLayout(LayoutKind.Explicit)]
+        internal struct Variant
+        {
+            [FieldOffset(0)] public TypeUnion m_Variant;
+            [FieldOffset(0)] public decimal m_decimal;
+        }
+
+#pragma warning disable 618
+        [Fact]
+        public static void NullParameter()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("pSrcNativeVariant", () => Marshal.GetObjectForNativeVariant(IntPtr.Zero));
+            AssertExtensions.Throws<ArgumentNullException>("pSrcNativeVariant", () => Marshal.GetObjectForNativeVariant<int>(IntPtr.Zero));
+        }
+
+        [DllImport(@"oleaut32.dll", SetLastError = true, CallingConvention = CallingConvention.StdCall)]
+        static extern Int32 VariantClear(IntPtr pvarg);
```

Comment:
This is still going to be problematic on Unix.  Either an equivalent for Unix should be provided, or the tests that use it (and maybe others?) should be made to run only on Windows by adding `[PlatformSpecific(TestPlatforms.Windows)]` to the relevant test(s).

Reply:
Scoped the 4 tests in this file which calls this only for Windows with `[PlatformSpecific(TestPlatforms.Windows)]`